clang/lib/Basic/DiagnosticIDs.cpp

```diff
@@ -286,6 +334,18 @@ DiagnosticMapping DiagnosticIDs::getDefaultMapping(unsigned DiagID) {
   return Info;
 }
 
+void DiagnosticIDs::initCustomDiagMapping(DiagnosticMapping &Mapping,
+                                          unsigned DiagID) {
+  assert(IsCustomDiag(DiagID));
+  const auto &Diag = CustomDiagInfo->getDescription(DiagID);
+  if (auto Group = Diag.GetGroup()) {
+    auto GroupInfo = GroupInfos[static_cast<size_t>(*Group)];
```

Comment:
Not a good use of `auto` because the type is not particularly obvious from the initialization.